File: devlibrary/src/main/java/com/example/devlibrary/network/exception/ApiException.java
package com.example.devlibrary.network.exception;

/**
 * Created by fanqh on 2017/10/26.
 */

public class ApiException extends RuntimeException {

    public int getErrorCode() {
        return errorCode;
    }

    public void setErrorCode(int errorCode) {
        this.errorCode = errorCode;
    }

    private int errorCode;

    public ApiException(int errorCode, String message) {
        super(message);
        this.errorCode = errorCode;
    }
}

File: baselib/src/main/java/com/wan/baselib/network/exception/ApiException.kt
package com.wan.baselib.network.exception

/**
 * Created by cy on 2017/10/26.
 */
class ApiException : RuntimeException {

    var errorCode = 0

    constructor(message: String?) : super(message)

    constructor(errorCode: Int, message: String?) : super(message) {
        this.errorCode = errorCode
    }

    constructor(message: String?, cause: Throwable?) : super(message, cause)
}